api_generator/lib/spec_hash.rb

```diff
@@ -59,8 +70,8 @@ def parse(value)
     return value unless value.is_a?(Hash)
     ref = value.delete('$ref')
     value.transform_values! { |v| parse(v) }
-    return SpecHash.new(value) unless ref
-    SpecHash.new(parse(resolve(ref)).merge(value))
+    value.merge!(resolve(ref)) if ref
```

Comment:
Generally modifying `value` in-place like this is not a good idea, duplicate and return value.

Reply:
I agree. This was done intentionally. The SpecHash is my poor-man OpenAPI parser (though 10 times faster than the official tool for ruby) that replaces all objects with `$ref` with the object it references. The original object will be discarded and not to be used by anything else. Hence the in-place update.